From d358aa98f1c82171af5afe815909a6d8ade0168a Mon Sep 17 00:00:00 2001
From: Lucian Petrut <lpetrut@cloudbasesolutions.com>
Date: Thu, 12 Sep 2024 13:58:24 +0000
Subject: [PATCH] quick hack: disable image set check

Calico currently checks if the image set that it received matches
the upstream images, which prevents us from using custom images.

This quick hack will just ignore unknown images. However, we should
use the Installation spec to parse the images, just like GetReference
does.
---
 pkg/controller/utils/imageset/imageset.go | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pkg/controller/utils/imageset/imageset.go b/pkg/controller/utils/imageset/imageset.go
index a825ddb9..b21aa693 100644
--- a/pkg/controller/utils/imageset/imageset.go
+++ b/pkg/controller/utils/imageset/imageset.go
@@ -122,7 +122,7 @@ func ValidateImageSet(is *operator.ImageSet) error {
                }
        }

-       if len(unknownImages) == 0 && len(invalidDigests) == 0 {
+       if len(invalidDigests) == 0 {
                return nil
        }

--
2.43.0
